src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -186,22 +196,30 @@ function CreateSimilarityIdxCtrl(
 
     $scope.saveSearchQuery = function () {
         $scope.saveOrUpdateExecuted = true;
-        if (!isDirty) {
```

Comment:
The purpose of this check is to avoid calling save if there are no changes in the index.

Reply:
Changed the logic to really prevent the save operation if the query is pristine.